packages/firebase_dynamic_links/firebase_dynamic_links/android/src/main/java/io/flutter/plugins/firebase/dynamiclinks/FlutterFirebaseDynamicLinksPlugin.java

```diff
@@ -105,21 +103,25 @@ static FirebaseDynamicLinks getDynamicLinkInstance(@Nullable Map<String, Object>
   }
 
   @Override
-  public boolean onNewIntent(Intent intent) {
+  public boolean onNewIntent(@NonNull Intent intent) {
     getDynamicLinkInstance(null)
         .getDynamicLink(intent)
         .addOnSuccessListener(
             pendingDynamicLinkData -> {
               Map<String, Object> dynamicLink =
                   Utils.getMapFromPendingDynamicLinkData(pendingDynamicLinkData);
-              if (dynamicLink != null) {
+              // We check if channel is `null` as this may be called after the plugin is detached from FlutterEngine
+              if (dynamicLink != null && channel != null) {
```

Comment:
Should we log something here?  Isn't it because the plugin is used in a weird state?

Reply:
Isn't the URL/link also going to be lost here, could there be a chance the link comes in while flutter is initialising still and this link is going to be lost?

This PR would fix the crash but I think there's probably more work to be done to maybe capture the link in this state and send it later when flutter is ready

Reply:
@Lyokone  Can log something if you wish, but I have cached the DL until the FlutterEngine is attached again, sending  the DL via `onLink` when app is ready so I'm not sure it's necessary anymore.

Reply:
LGTM thanks :)